src/Microsoft.Health.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -197,31 +198,38 @@ private bool CanInitialize()
 
             if (_sqlServerDataStoreConfiguration.AllowDatabaseCreation)
             {
-                var connection = GetConnectionIfDatabaseNotExists(_sqlServerDataStoreConfiguration.ConnectionString, databaseName);
-
-                if (connection != null)
+                using (var connection = _sqlConnection.GetSqlConnection(connectToMaster: true))
                 {
-                    _logger.LogInformation("Database does not exist");
-
-                    bool created = CreateDatabase(connection, databaseName);
+                    bool doesDatabaseExist = DoesDatabaseExist(connection, databaseName);
 
-                    if (created)
+                    if (!doesDatabaseExist)
                     {
-                        _logger.LogInformation("Created database");
-                    }
-                    else
-                    {
-                        _logger.LogWarning("Insufficient permissions to create the database");
-                        return false;
-                    }
+                        _logger.LogInformation("Database does not exist");
+
+                        bool created = CreateDatabase(connection, databaseName);
+
+                        if (created)
+                        {
+                            _logger.LogInformation("Created database");
+                        }
+                        else
+                        {
+                            _logger.LogWarning("Insufficient permissions to create the database");
+                            return false;
+                        }
 
-                    connection.Close();
+                        connection.Close();
```

Comment:
Since this is now within a using block, I don't believe the `connection.Close()` needs to be directly called anymore.

Reply:
I just double checked. What you are stating is absolutely correct. Will update in the next iteration.